tests/bsim/bluetooth/host/l2cap/reassembly/dut/src/dut.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/logging/log.h>
+
+#include "testlib/att_read.h"
+#include "testlib/att_write.h"
+#include "testlib/conn.h"
+#include "testlib/scan.h"
+#include "testlib/log_utils.h"
+
+#include "babblekit/flags.h"
+#include "babblekit/testcase.h"
+
+/* local includes */
+#include "data.h"
+
+LOG_MODULE_REGISTER(dut, LOG_LEVEL_DBG);
+
+extern unsigned long runtime_log_level;
+
+static DEFINE_FLAG(got_notification);
+
+static uint8_t received_notification(struct bt_conn *conn,
+				     struct bt_gatt_subscribe_params *params,
+				     const void *data,
+				     uint16_t length)
+{
+	if (length) {
+		LOG_INF("RX notification");
+		LOG_HEXDUMP_DBG(data, length, "payload");
+
+		SET_FLAG(got_notification);
+	}
+
```

Comment:
We should validate the content of the notification we got.

Reply:
should we? I guess I can, it makes the test stricter. 
The purpose was just checking for leaks though.

Reply:
I have no strong opinion here. I just though it made sense to add some simple, cheap validation here.